torch/fx/passes/shape_prop.py

```diff
@@ -182,4 +182,6 @@ def propagate(self, *args):
         Returns:
             Any: The value returned from executing the Module
         """
-        return super().run(*args)
+        # clone inputs to avoid side effects caused by inplace ops during run_node
+        new_args = [torch._prims_common.clone_preserve_strides(x) for x in args]
```

Comment:
This line in particular is problematic when attempting to use with torchrec's [KeyedJaggedTensor](https://github.com/pytorch/torchrec/blob/3138495d1b3a6d3cdb0561c802f2a3994dd3d24e/torchrec/sparse/jagged_tensor.py#L682), resulting in errors like:

```
in clone_preserve_strides
    x.size(), x.stride(), x.storage_offset()
AttributeError: 'KeyedJaggedTensor' object has no attribute 'size'
```

Reply:
Hi @seemethere I am curious about what case will trigger this issue. I added clone_preserve_strides in two places.

1. In this function. As the function description said, *args should be `Tensor`. I think Tensor must have attribute 'size', right? 

```
def propagate(self, *args):
        """
        Run `module` via interpretation and return the result and
        record the shape and type of each node.
        Args:
            *args (Tensor): the sample input.
        Returns:
            Any: The value returned from executing the Module
        """
```

2. Similarly, in function aot_dispatch_base, flat_args should be `List[Tensor]`.

`def aot_dispatch_base(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig):`